pkg/operator/controller/externaldns/status_test.go

```diff
@@ -0,0 +1,668 @@
+			existingDeployment: fakeDeployment(appsv1.DeploymentAvailable, corev1.ConditionTrue, 8, "0%", "0%", 8),
+			expectedResult: metav1.Condition{
+				Type:               ExternalDNSDeploymentReplicasMinAvailableConditionType,
+				Status:             metav1.ConditionTrue,
+				Reason:             "DeploymentMinimumReplicasMet",
+				Message:            "Minimum replicas requirement is met",
+				LastTransitionTime: metav1.NewTime(clock.Now())},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			cond := computeMinReplicasCondition(&tc.existingDeployment)
+			require.EqualValues(t, tc.expectedResult.Type, cond.Type)
+			require.EqualValues(t, tc.expectedResult.Status, cond.Status)
+		})
+	}
+}
+
+func TestComputeAllReplicasCondition(t *testing.T) {
+	testCases := []struct {
+		name               string
+		existingDeployment appsv1.Deployment
+		expectedResult     metav1.Condition
+	}{
+		{
+			name:               "AvailableReplica = spec.replica should return ConditionTrue",
+			existingDeployment: fakeDeployment(appsv1.DeploymentAvailable, corev1.ConditionTrue, 8, "25%", "25%", 8),
+			expectedResult: metav1.Condition{
+				Type:               ExternalDNSDeploymentReplicasAllAvailableConditionType,
+				Status:             metav1.ConditionTrue,
+				Reason:             "DeploymentReplicasAvailable",
+				Message:            "All replicas are available",
+				LastTransitionTime: metav1.NewTime(clock.Now()),
+			},
+		},
+		{
+			name:               "AvailableReplica < spec.replica shoud return ConditionFalse",
+			existingDeployment: fakeDeployment(appsv1.DeploymentProgressing, corev1.ConditionFalse, 8, "25%", "25%", 6),
+			expectedResult: metav1.Condition{
+				Type:               ExternalDNSDeploymentReplicasAllAvailableConditionType,
+				Status:             metav1.ConditionFalse,
+				Reason:             "DeploymentReplicasNotAvailable",
+				Message:            "Irrelevant for test",
+				LastTransitionTime: metav1.NewTime(clock.Now()),
+			},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			cond := computeAllReplicasCondition(&tc.existingDeployment)
+			require.EqualValues(t, tc.expectedResult.Type, cond.Type)
+			require.EqualValues(t, tc.expectedResult.Status, cond.Status)
+		})
+	}
+}
+
+func TestComputeDeploymentPodsScheduledCondition(t *testing.T) {
+	testCases := []struct {
+		name               string
+		existingDeployment appsv1.Deployment
+		existingPods       []corev1.Pod
+		expectedResult     metav1.Condition
+	}{
+		//all podScheduledConditions are true
+		{
+			name: "All pods are scheduled should return ConditionTrue",
+			existingDeployment: appsv1.Deployment{
+				Spec: appsv1.DeploymentSpec{
+					Selector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"name": "external-dns-operator",
+						},
+					},
+				},
+			},
+			existingPods: []corev1.Pod{
```

Comment:
Can you use a function similar to `fakeDeployment` for PODs too?
This would save a lot of space and make test more readable.

Reply:
Done. to be checked and resolved